src/examples/src/widgets/tab-controller/Closeable.tsx

```diff
@@ -1,39 +1,33 @@
 import { tsx, create } from '@dojo/framework/core/vdom';
-import icache from '@dojo/framework/core/middleware/icache';
-import { includes } from '@dojo/framework/shim/array';
 
-import TabController from '@dojo/widgets/tab-controller';
-import Tab from '@dojo/widgets/tab';
+import TabController, { TabContent } from '@dojo/widgets/tab-controller';
 
-const factory = create({ icache });
+const factory = create();
+
+export default factory(function Closeable() {
+	const tabs = [
+		{ closeable: true, label: 'Tab One' },
+		{ label: 'Tab Two' },
+		{ label: 'Tab Three' },
+		{ label: 'Tab Four' }
+	];
 
-export default factory(function Closeable({ middleware: { icache } }) {
-	const activeIndex = icache.getOrSet('active', 0);
-	const closedKeys = icache.getOrSet<string[]>('closedKeys', []);
 	return (
-		<TabController
-			activeIndex={activeIndex}
-			onRequestTabChange={(index) => {
-				icache.set('active', index);
-			}}
-			onRequestTabClose={(index, key) => {
-				icache.set('closedKeys', [...closedKeys, key]);
-			}}
-		>
-			{!includes(closedKeys, 'tab-one') ? (
-				<Tab key="tab-one" label="Tab One" closeable={true}>
+		<TabController tabs={tabs}>
+			{(_tabs, isActive, isClosed) => [
+				<TabContent key="tab0" active={isActive(0)} closed={isClosed(0)}>
 					Hello Tab One
-				</Tab>
-			) : null}
-			<Tab key="tab-two" label="Tab Two">
-				Hello Tab Two
-			</Tab>
-			<Tab key="tab-three" label="Tab Three">
-				Hello Tab Three
-			</Tab>
-			<Tab key="tab-four" label="Tab Four">
-				Hello Tab Four
-			</Tab>
+				</TabContent>,
+				<TabContent key="tab1" active={isActive(1)} closed={isClosed(1)}>
+					Hello Tab Two
+				</TabContent>,
```

Comment:
Note that `active()` and `closed()` take numerical indices instead of string keys, since otherwise `<TabController>` would need to be able to map each index to a key in order to change the active index/closed indices. If need be, we could change this by requiring each `TabItem` config to have a string `key` that would be used within the child renderer.

Reply:
if the config took anything, it should be `id` rather than `key` as key is such an overloaded term.

Reply:
I updated `TabItem` with a required `id: string` property, and updated the methods passed to the child renderer.